### ISSUE TITLE
[BUG] Cannot deserialize Microsoft.Azure.Search.Models.SearchResult

### ISSUE THREAD

**Issue by rcbevans** [Azure/azure-sdk-for-net#9927]
**Describe the bug**
The latest version of the the `Microsoft.Azure.Search.Models.SearchResult` class has Document with no public setter, and no JsonProperty attribute for deserialization.  As a result, the Document property cannot be deserialized.  Score and Highlights deserialize correctly.

We return a list of Microsoft.Azure.Search.Models.SearchResult<T> from an API, and our integration tests deserialize the returned JSON to the C# type to validate equivalence.

**Expected behavior**
Microsoft.Azure.Search.Models.SearchResult should be deserializable from JSON, as well as serializable.

**Actual behavior (include Exception or Stack Trace)**
The Document property is `null` after deserialization.

**To Reproduce**

```C#
    public class TestClass
    {
        public TestClass(string testString)
        {
            this.TestString = testString;
        }

        public TestClass()
        {
        }

        public string TestString { get; private set; }
    }


    public class Program
    {
        public static void Main(string[] args)
        {
            var test = new List<SearchResult<TestClass>>()
            {
                new SearchResult<TestClass>(new TestClass("Foo"), 45.0),
                new SearchResult<TestClass>(new TestClass("Bar"), 22.7)
            };

            var serialized = JsonConvert.SerializeObject(test, new JsonSerializerSettings() { ContractResolver = new DefaultContractResolver() });
            var deserialized = JsonConvert.DeserializeObject<List<SearchResult<TestClass>>>(serialized, new JsonSerializerSettings() { ContractResolver = new DefaultContractResolver() });

            Console.WriteLine(test.ToString());
            Console.WriteLine(serialized);
            Console.WriteLine(deserialized);
        }
    }
```

**Environment:**
Microsoft.Azure.Search 10.1.0
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): Windows 10 .Net Core 3.1 in ASP.NET Core 3.1 web project
 - IDE and version : Visual Studio 2019 16.4.5

**Comment by jsquire** [comment]
//cc: @tg-msft, @AlexGhiondea

**Comment by brjohnstmsft** [comment]
@rcbevans What you're trying to do is not a supported scenario, but you might be able to achieve it if you use the `DeserializationSettings` property of `SearchIndexClient` when calling `JsonConvert`. Let me know if this workaround unblocks you.

That said, I would advise you not to take a dependency on the SDK's model classes in your own API surface. That's a degree of coupling that will come back to haunt you later, because these model classes **will** change.

**Comment by rcbevans** [comment]
Yeah, I understand.  My preference would be us use Automapper and have DTOs on the API surface, but this code predates me; I'm just doing the work to upgrade the platform my team is using.  I was able to achieve the same end result and get the tests passing my defining my own TestSearchResult with properties accessible properties mirroring the Azure type, and use FluentAssertions Should().BeEquivalentTo() to verify equivalence.

**Comment by brjohnstmsft** [comment]
@rcbevans Glad to hear you're unblocked. I'm closing this issue since it's not a bug per se. If external serialization of models is something you think Azure .NET SDKs should support, please open a new issue to track it as a feature request.

**Comment by dauv** [comment]
@brjohnstmsft Why is this not considered a bug? We have implemented a Redis cache layer to avoid throttling issues, but above issue causes the document to always be null. The proposed workaround doesn't work for me, I guess the only way around it is to depend on the SDK models?

**Comment by brjohnstmsft** [comment]
@dauv The model classes for `Microsoft.Azure.Search` were designed in tandem with custom serialization logic specific to that library; There was never an intention to support serialization/deserialization of these types by 3rd parties. That's what I meant earlier when I said this was an unsupported scenario. We made no promise that this would work, so I view it more as an intentionally missing feature than a bug.

> I guess the only way around it is to depend on the SDK models?

It's the opposite -- write your own models instead of depending on the SDK models. That approach worked for @rcbevans.

That said, I'm curious why you need to represent the search result payload as strongly-typed objects in a caching layer. Do you have code that manipulates the results before returning them to the client?

Also please note that `Microsoft.Azure.Search` has been deprecated; Please consider upgrading to [Azure.Search.Documents](https://www.nuget.org/packages/Azure.Search.Documents) if possible. There's a migration guide [here](https://docs.microsoft.com/en-us/azure/search/search-dotnet-sdk-migration-version-11).
